public_html/wp-content/themes/pattern-directory/src/components/my-patterns/index.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { addQueryArgs } from '@wordpress/url';
+import { store as coreStore } from '@wordpress/core-data';
+import { store as patternStore } from '../../store';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import Menu from './menu';
+import PatternGrid from '../pattern-grid';
+import QueryMonitor from '../query-monitor';
+import { RouteProvider } from '../../hooks';
+
+const MyPatterns = () => {
+	const query = useSelect( ( select ) => select( patternStore ).getCurrentQuery() );
+	const author = useSelect( ( select ) => select( coreStore ).getCurrentUser()?.id );
+
+	if ( ! author ) {
+		const loginUrl = addQueryArgs( wporgLoginUrl, { redirect_to: window.location } );
+		return (
+			<div className="entry-content">
+				<p>{ __( 'Please log in to view your patterns.', 'wporg-patterns' ) }</p>
+				<a className="button button-primary" href={ loginUrl }>
+					{ __( 'Log in', 'wporg-patterns' ) }
+				</a>
+			</div>
+		);
+	}
+	// Show all patterns regardless of status, but if the current query has a status (the view is draft, for
+	// example), that will override `any`. Lastly, make sure this shows the current user's patterns.
+	const modifiedQuery = { status: 'any', ...query, author: author };
+
+	return (
+		<RouteProvider>
+			<QueryMonitor />
+			<Menu />
```

Comment:
The menu on mobile doesn't show the selected category by default because its original usage was supplemented by the context bar. Now that there isn't one here, it's hard to tell which category is selected.

Could we maybe append it like "Menu: All" on mobile?

Currently:
![](https://d.pr/i/VcoWD6.png)

Reply:
Once #164 is done, each pattern will have a status flag, so it should be more obvious. Adding the current item into the label works visually, but might be confusing for screen reader users - for example if the menu will be labelled "Menu: Drafts", it sounds like it might be a menu for drafts.